src/Resource.ts

```diff
@@ -55,7 +56,16 @@ export class Resource {
     options.method = options.method || "get"
     options.requestConfig = options.requestConfig || {}
     options.property = options.property || null
-
+    let headersFn: Function
```

Comment:
Could you please extract the logic in an separate private method and just do sth. like 
`headersFn = getHeadersFn(options.headers)` ?